assignments/array-methods.js

```diff
@@ -2,82 +2,531 @@
+// The event director needs both the first and last names of each runner for their running bibs.
+// Combine both the first and last names into a new array called fullName.
 let fullName = [];
+runners.forEach(function(person) {
+	let name = person.first_name + ' ' + person.last_name;
+	fullName.push(name);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
+// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER.
+// Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+runners.map(function(person) {
+	let name = person.first_name.toUpperCase();
+	allCaps.push(name);
+});
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+// The large shirts won't be available for the event due to an ordering issue.
+// Get a list of runners with large sized shirts so they can choose a different size.
+// Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+let onlyLargeRunners = runners.filter(checkIfLarge);
+function checkIfLarge(runner) {
+	if (runner.shirt_size === 'L') {
+		largeShirts.push(runner);
+	}
+}
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
+// The donations need to be tallied up and reported for tax purposes.
+// Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+ticketPriceTotal = runners.reduce(adder, 0);
+function adder(acc, runner) {
+	return acc + runner.donation;
+}
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+// Return a list of all the small businesses and ids
+let companyInfo = [];
+runners.forEach(function(person) {
+	companyInfo.push(person.id, person.company_name);
+});
+console.log(companyInfo);
 
 // Problem 2
+// Bussiness that donated 150 or more
+let bigDonations = [];
+let bigDonnors = runners.filter(checkDonation);
```

Comment:
The filter() method creates an array filled with all array elements that pass a test (provided as a function).

check above comments.